`mypy` fails due to upgraded narwhals type declarations

```py
src/bokeh/core/property/data_frame.py: note: In member "validate" of class "EagerDataFrame":                                   
src/bokeh/core/property/data_frame.py:63:13: error: Statement is unreachable  [unreachable]                                    
                return                                                                                                         
                ^~~~~~                                                                                                         
src/bokeh/core/property/data_frame.py: note: In member "validate" of class "EagerSeries":                                      
src/bokeh/core/property/data_frame.py:81:13: error: Statement is unreachable  [unreachable]                                    
                return                                                                                                         
                ^~~~~~                                                                                                         
Found 2 errors in 1 file (checked 658 source files) 
```

--- a/src/bokeh/core/property/data_frame.py
+++ b/src/bokeh/core/property/data_frame.py
@@ -60,7 +60,7 @@ class EagerDataFrame(Property["IntoDataFrame"]):
         super().validate(value, detail)
 
         if nw.dependencies.is_into_dataframe(value):
-            return
+            return # type: ignore[unreachable] # https://github.com/bokeh/bokeh/issues/14342
 
         msg = "" if not detail else f"expected object convertible to Narwhals DataFrame, got {value!r}"
         raise ValueError(msg)
@@ -78,7 +78,7 @@ class EagerSeries(Property["IntoSeries"]):
         super().validate(value, detail)
 
         if nw.dependencies.is_into_series(value):
-            return
+            return # type: ignore[unreachable] # https://github.com/bokeh/bokeh/issues/14342
 
         msg = "" if not detail else f"expected object convertible to Narwhals Series, got {value!r}"
         raise ValueError(msg)

